### PR TITLE
Added code coverage checker

### PR DESCRIPTION
**Pull request by allenhsieh**
Added Istanbul (not Constantinople) as a code coverage checker. This PR is carrying over #141 which cannot merge to master.

This pull request makes the following changes:
* Added Istanbul as code coverage checker
* Updated scripts and circleci yml to run tests with coverage
* Refactored tests because `nock` can be chained
* Modified logger behavior so that errors are written to stdout as well

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,10 @@
     "test": "mocha -r ts-node/register src/**/*.spec.ts",
     "build": "tsc -p tsconfig.json",
     "build-dev": "tsc -p tsconfig.development.json",
+    "coverage": "nyc mocha -r ts-node/register src/**/*.spec.ts",
+    "generate-coverage-report": "nyc report --reporter=text-lcov > lcov.info",
+    "test-ci": "MOCHA_FILE=./reports/test-results.xml mocha -r ts-node/register src/**/*.spec.ts --reporter mocha-junit-reporter",
+    "test-console": "mocha -r ts-node/register src/**/*.spec.ts --reporter mocha-junit-reporter --reporter-options toConsole",
```
Comment (DarthHater) [inline package.json]: I would remove this, as I think it was just for local testing.

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,8 @@
 auditjs.json
 # ci config for local ci build
 /.circleci/local-config.yml
+# Instanbul code coverage report
+.nyc_output/
+reports/
+lcov.info
+test-results.xml
```
Comment (DarthHater) [inline .gitignore]: Trivial: newline

File: package.json (hunk 1 above)
Comment (allenhsieh) [reply package.json]: in: https://github.com/sonatype-nexus-community/auditjs/pull/173/commits/8400685999c790751b590ae174f33cb14f3f7788

**Comment by DarthHater** [comment]
Whoa you can chain nock?!?!

@allenhsieh u r teh wizard

**Pull request merged**